website/blog/2022-12-13-pointer-events-in-react-native.md

```diff
@@ -0,0 +1,175 @@
+
+### Event Properties
+
+For each of the events mentioned above we’ve also implemented the majority of the properties expected in the PointerEvent object — though in React Native these are exposed through the `event.nativeEvent` property. You can find an enumeration of all the implemented properties in the [event object’s Flowtype interface definition](https://github.com/facebook/react-native/blob/59ee57352738f030b41589a450209e51e44bbb06/Libraries/Types/CoreEventTypes.js#L175). One notable exception to being completely implemented is the `relatedTarget` property as exposing a native view reference in this ad-hoc manner isn’t trivial.
+
+## Future Work and Exploration
+
+In addition to the events above there are also some other APIs related to Pointer Events. In the future, we plan to be implement these APIs as a part of this effort. These APIs include:
+
+- Pointer Capture API
+  - Includes the imperative API exposed on element references including `setPointerCapture()`, `releasePointerCapture()`, and `hasPointerCapture()`.
+- `touch-action` style property
+  - The web uses this CSS property to declaratively negotiate gestures between the browser and a website’s own event handling code. In React Native this could be used for negotiating the event handling between a View’s pointer event handlers and a parent ScrollView.
+- `click`, `contextmenu`, `auxclick`
+  - `click` is an abstract definition of interaction that may be triggered through accessibility paradigms or other characteristic platform interactions.
+
+Another benefit of the native Pointer Events implementation is that it will allow us to revisit and improve various forms of gesture handling currently limited only to touch events and handled in JavaScript by the Responder, Pressability, and PanResponder APIs.
+
+Furthermore, we are continuing to explore including an implementation of the `EventTarget` interface for React Native host components (i.e. `add`/`removeEventListener`) which we believe will make possible more user-land abstractions for handling pointer interactions.
+
+## Trying it Out
+
+Our Pointer Events implementation is still experimental but we’re interested in getting feedback from the community on what we’ve shared. If you are interested in trying this API you’ll need to enable a couple feature flags:
+
+### Enable Feature Flags
+
+Note: Pointer Events are only implemented for the new architecture and are only available for React Native 0.71+ which at the time of writting is a release candidate.
+
+In your entry JavaScript file (index.js in the default React Native app template) you’ll need to enable the `shouldEmitW3CPointerEvents` flag for Pointer Events and `shouldPressibilityUseW3CPointerEventsForHover` to use Pointer Events in `Pressability`.
+
+```js
+import ReactNativeFeatureFlags from 'react-native/Libraries/ReactNative/ReactNativeFeatureFlags';
```

Comment:
Is there any reason we couldn't export this from the RN entry-point? This line will cause a typechecking error in TS (only certain internal modules are given types), and I think we are wanting to eventually prevent libs from reaching in.